repos/iiif-storage/src/JsonBuilder/MetadataAggregator.php

```diff
@@ -2,54 +2,63 @@
 
 namespace IIIFStorage\JsonBuilder;
 
+use Digirati\OmekaShared\Utility\OmekaValue;
 use LogicException;
 use Omeka\Api\Representation\ItemRepresentation;
 use Omeka\Api\Representation\ItemSetRepresentation;
 use Omeka\Api\Representation\ValueRepresentation;
 
 trait MetadataAggregator
 {
-    public function aggregateMetadata($representation, &$json) {
+    public function aggregateMetadata($representation, &$json)
+    {
         if (!$representation instanceof ItemRepresentation && !$representation instanceof ItemSetRepresentation) {
             throw new LogicException('Items and ItemSets can be aggregated only.');
         }
         $mapping = $this->getFunctionalFields();
         $blacklist = $this->getBlacklist();
 
-        foreach ($representation->values() as $key => $value) {
+        foreach ($representation->values() as $key => $values) {
             if (in_array($key, $blacklist)) {
                 continue;
             }
             /** @var ValueRepresentation $value */
             $mappingField = $mapping[$key] ?? null;
-            $value = $representation->value($key);
             if ($mappingField) {
-                $json[$mappingField] = $value->value();
+                $json[$mappingField] = OmekaValue::toRdf($representation, $key);
             } else {
-                switch ($value->type()) {
-                    case 'uri':
-                        $json['metadata'][] = [
-                            'label' => $value->property()->label(),
-                            'value' => $value->asHtml(),
-                        ];
-                        break;
-                    case 'literal':
-                        $json['metadata'][] = [
-                            'label' => $value->property()->label(),
-                            'value' => $value->value(),
-                        ];
-                        break;
-                    default:
-                        break;
+                // @todo Pass the labels through the translator in Omeka
```

Comment:
Hm, do we need to export vocabs to the external translation tool?

Reply:
Omeka 1.4 added [something that might help](https://github.com/omeka/omeka-s/pull/1373) but this appears to be just for displaying the vocabulary. It's possible we might need to do this, but I don't think it is worth doing at the moment. Specifically, the labels I'm referring to in this `@todo` are the ones from the resource templates:
<img width="1043" alt="Screenshot 2019-04-11 at 19 40 52" src="https://user-images.githubusercontent.com/8266711/55982971-ebf18c80-5c91-11e9-8808-00b139fad2fd.png">

Which is used in the metadata key-value pair on an omeka item. (bad example here showing `sc:hasManifests` but imagine another RDF term)

So maybe we just need to export the resource templates, and then pass them through translate in the template

Reply:
Ah yeah, that could be as simple as exporting message keys for `dcterms:identifier` or something.